docs/DeveloperGuide.md

```diff
@@ -133,85 +134,50 @@ Classes used by multiple components are in the `seedu.patientbook.commons` packa
+5. The `DoneCommandParser` creates a new `DoneCommand` instance with the newly created `DateTimeLoader` object and returns it to `NuudleParser`, which in turn returns it to `LogicManager`.
 
-![UndoRedoState1](images/UndoRedoState1.png)
+6. `LogicManager` calls the `DoneCommand#execute(model)` method.
 
-Step 3. The user executes `add n/David …​` to add a new patient. The `add` command also calls `Model#commitPatientBook()`, causing another modified patient book state to be saved into the `patientBookStateList`.
+7. `DoneCommand` obtains a copy of the `FilteredAppointmentList` by calling the `Model#getFilteredAppointmentList()` method. 
 
-![UndoRedoState2](images/UndoRedoState2.png)
+8. `DoneCommand` returns the appointment `toMark` in the `FilteredAppointmentList`, if there is an appointment in the list starts at the same time with the date and time indicated in the `DateTimeLoader`; Otherwise, throw an 
+`APPOINTMENT_DOES_NOT_EXISTS` exception.
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If a command fails its execution, it will not call `Model#commitPatientBook()`, so the patient book state will not be saved into the `patientBookStateList`.
+9. `DoneCommand` creates another instance of this appointment `doneAppointment` which has a `done` status. 
 
-</div>
-
-Step 4. The user now decides that adding the patient was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoPatientBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous patient book state, and restores the patient book to that state.
-
-![UndoRedoState3](images/UndoRedoState3.png)
-
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index 0, pointing to the initial PatientBook state, then there are no previous PatientBook states to restore. The `undo` command uses `Model#canUndoPatientBook()` to check if this is the case. If so, it will return an error to the user rather
-than attempting to perform the undo.
-
-</div>
-
-The following sequence diagram shows how the undo operation works:
-
-![UndoSequenceDiagram](images/UndoSequenceDiagram.png)
-
-<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
-
-</div>
+10. `DoneCommand` replaces the `toMark` with the `doneAppointment` by calling the `Model#setAppointment(target, editedAppointment)`.
 
-The `redo` command does the opposite — it calls `Model#redoPatientBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the patient book to that state.
+11. As a last step, `DoneCommand` creates a `CommandResult` with `SuccessMessage` and returns it to `LogicManager`.
```

Comment:
Maybe it'll improve the flow by replacing the phrase "As a last step" with "Lastly,".

Also, it seems to me that you missed out a "a" in  "`CommandResult` with a `SuccessMessage` ".